object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -350,340 +302,21 @@ public ObjectConstructionTypeAnnotator(AnnotatedTypeFactory atypeFactory) {
     public Void visitExecutable(AnnotatedTypeMirror.AnnotatedExecutableType t, Void p) {
       ExecutableElement element = t.getElement();
 
-      String methodName = element.getSimpleName().toString();
-
       TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
-      TypeMirror superclass = enclosingElement.getSuperclass();
-
-      if ("toBuilder".equals(methodName)) {
-        if (hasAnnotation(enclosingElement, AutoValue.class)
-            && element.getModifiers().contains(Modifier.ABSTRACT)) {
-          handleToBuilder(t, enclosingElement, BuilderKind.AUTO_VALUE);
-          return super.visitExecutable(t, p);
-        }
-        // check superclass, to handle generated code
-        if (!superclass.getKind().equals(TypeKind.NONE)) {
-          TypeElement superElement = TypesUtils.getTypeElement(superclass);
-          if (hasAnnotation(superElement, AutoValue.class)) {
-            handleToBuilder(t, superElement, BuilderKind.AUTO_VALUE);
-            return super.visitExecutable(t, p);
-          }
-        }
 
-        if (hasAnnotation(element, "lombok.Generated")
-            || hasAnnotation(enclosingElement, "lombok.Generated")) {
-          handleToBuilder(t, enclosingElement, BuilderKind.LOMBOK);
-        }
+      for (FrameworkSupport frameworkSupport : frameworkSupports) {
+        frameworkSupport.handleToBuilder(t);
```

Comment:
Should this method also have a name that indicates that it handles *possible* toBuilder methods (like `handlePossibleBuilderBuildMethod `)? I think we should be consistent across the framework support about method naming.